source/server/admin/active_stats.js

```diff
@@ -0,0 +1,192 @@
+}
+
+/**
+ * Function used to sort stat records. The highest priority is update frequency,
+ * then value (higher values are likely more interesting), and finally alphabetic,
+ * in forward order.
+ *
+ * @param {!Object} a
+ * @param {!Object} b
+ * @return {number}
+ */
+function compareStatRecords(a, b) {
+  // Sort higher change-counts first.
+  if (a.change_count != b.change_count) {
+    return b.change_count - a.change_count;
+  }
+
+  // Secondarily put higher values first -- they are often more interesting.
+  if (b.value != a.value) {
+    return b.value - a.value;
+  }
+
+  // Fall back to forward alphabetic sort.
+  if (a.name < b.name) {
+    return -1;
+  }
+  if (a.name > b.name) {
+    return 1;
+  }
+  return 0;
+}
+
+/**
+ * The active display has additional settings for tweaking it -- this helper extracts numeric
+ * values from text widgets
+ *
+ * @param {string} id
+ * @param {number} defaultValue
+ * @return {number}
+ */
+function loadSettingOrUseDefault(id, defaultValue) {
+  const elt = document.getElementById(id);
+  const value = parseInt(elt.value);
+  if (Number.isNaN(value) || value <= 0) {
+    console.log('Invalid ' + id + ': invalid positive number');
+    return defaultValue;
+  }
+  return value;
+}
+
+/**
+ * Rendering function which interprets the Json response from the server, updates
+ * the most-frequently-used map, reverse-sorts by use-count, and serializes the
+ * top ordered stats into the PRE element created in initHook.
+ *
+ * @param {!Object} data
+ */
+function renderStats(data) {
+  sortedStats = [];
+  let prevStats = currentStats;
+  currentStats = new Map();
+  for (stat of data.stats) {
+    if (!stat.name) {
+      continue; // Skip histograms for now.
+    }
+    let statRecord = prevStats.get(stat.name);
+    if (statRecord) {
+      if (statRecord.value != stat.value) {
+        statRecord.value = stat.value;
+        ++statRecord.change_count;
+      }
+    } else {
+      statRecord = {name: stat.name, value: stat.value, change_count: 0};
+    }
+    currentStats.set(stat.name, statRecord);
+    sortedStats.push(statRecord);
+  }
+  prevStats = null;
+
+  // Sorts all the stats. This is inefficient; we should just pick the top N
+  // based on field "active-max-display-count" and sort those. The best
+  // algorithms for this require a heap or priority queue. JS implementations
+  // of those can be found, but that would bloat this relatively modest amount
+  // of code, and compell us to do a better job writing tests.
+  sortedStats.sort(compareStatRecords);
+
+  const max = loadSettingOrUseDefault('active-max-display-count', 50);
+  let index = 0;
+  let text = '';
+  for (const statRecord of sortedStats) {
+    if (++index > max) {
+      break;
+    }
+    text += statRecord.name + ': ' + statRecord.value + ' (' +
+        statRecord.change_count + ')' + '\n';
+  }
+  activeStatsPreElement.textContent = text;
```

Comment:
This kind of interface can be frustrating, if e.g. the user wants to copy and paste some content, erasing and re-setting the content every 5 seconds makes UI selection awkward.

This can potentially be worked around by making it a block of spans instead of a single text block, then rearranging the existing spans where the new data is similar to the old data, though that can be a bit complicated (kind of like writing a whole `diff`, but since you're only working with double-digit, maybe triple-digit amounts of content you can brute force it.)

Or by including a pause/restart button that just aborts an existing setTimeout / starts a new one.

Or it's reasonable to just deem this unimportant because it's not an important user-facing interface.